src/libs/WorkspacesSettingsUtils.ts

```diff
@@ -175,12 +180,18 @@ function getWorkspacesBrickRoads(): Record<string, BrickRoad> {
         }
     });
 
-    Object.values(allReports).forEach((report) => {
+    const actionsByReport: OnyxCollection<ReportActions> = {};
+    Object.keys(reportActions ?? {}).forEach((key) => {
+        const reportID = CollectionUtils.extractCollectionItemID(key as typeof ONYXKEYS.COLLECTION.REPORT_ACTIONS);
```

Comment:
Why are we extracting the ID from the key if we have it in the value that we already have?

Reply:
Because the forEach `key` looks like this: `reportActions_2298475481263683` but in order to create the object needed by the `getBrickRoadForPolicy` function, we need to extract the ID from the `key`, same as we're doing here:

https://github.com/Expensify/App/blob/ab8c6d3f7b4e73673d2bcaa5e487a3ef3c4f7cfd/src/libs/WorkspacesSettingsUtils.ts#L63-L64

Reply:
I know that. I am saying that each entry in `reportActions` has a key with the ID already, so you don't need to call `extractCollectionItemID` on its key to get that data, you can get it directly from the reportAction object.

Reply:
I see what you mean. If you're asking why am I doing it like this: because it works and it's used above in the component in similar fashion.

If you propose another way of writing it, please share it and I'll replace it.

Reply:
Not sure if we use underscore still but `_.pluck(reportActions, 'reportActionID')`

Reply:
As per [TS - Migration Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/TS_STYLE.md#migration-guidelines):

> Deprecate the usage of `underscore`. **Use vanilla methods from JS instead**. Only use `lodash` when there is no easy vanilla alternative (eg. lodashMerge). eslint: [no-restricted-imports](https://eslint.org/docs/latest/rules/no-restricted-imports)

This is why I went with `Object.keys(reportActions ?? {}).forEach`.

Reply:
But is the bug present in that call, given we are not passing the actions?

Reply:
No - if you're referring to the `getBrickRoadForPolicy` function discussed [here](https://github.com/Expensify/App/pull/40197#discussion_r1571181353), this is used only in one component BottomTabBar to display RBR dot above the Chats icon:

<img width="123" alt="Screenshot 2024-04-18 at 21 30 21" src="https://github.com/Expensify/App/assets/56457735/382b3fed-8ae2-4f09-8743-1473e936a08a">

Here's a video showing that it works as expected without any changes, even when refreshing the page (our issue with the workspace switcher):

https://github.com/Expensify/App/assets/56457735/abc626b8-2252-490b-a070-862faaa0d93c

cc @iwiznia